### PR TITLE
Add context introspection API to wrapper

### LINKED ISSUE #146: `wrapper.context()` API
Similar to [`wrapper.props`](https://github.com/airbnb/enzyme/blob/master/docs/api/ShallowWrapper/props.md), there should be an api to introspect into the context of a component. 

The vernacular is sort of weird if we want to mimic the `prop`/`props` style as it would be:

```js
wrapper.contexts() // returns entire context object
wrapper.context('foo') // returns foo from the context object
```

Part of me thinks the word `contexts` sounds weird. We could do this for an API:

```js
wrapper.context() // no argument - returns entire context object
wrapper.context('foo') // returns foo from the context object
```

though this might cause some confusion and unreliability to the API. Thoughts?

### PR DESCRIPTION
**Pull request by blainekasten**
### Fixes #146.

Adds an API to shallow/mount wrappers called `context` based on the discussion in the above thread. The below now works.

```js
wrapper.context() // returns context object
wrapper.context('foo') // returns foo off of context object
```

### CODE REVIEW

**Comment by blainekasten** [comment]
I suppose I probably need to add documentation for this too.

File: docs/api/ShallowWrapper/state.md [hunk 1]
```diff
@@ -25,3 +25,5 @@ expect(wrapper.state('foo')).to.equal(10);
 
 - [`.props() => Object`](props.md)
 - [`.prop(key) => Any`](prop.md)
+- [`.context(key) => Any`](context.md)
+- [`.context(key) => Any`](context.md)
```
Comment (lelandrichardson) [inline docs/api/ShallowWrapper/state.md]: duplicate?

File: docs/api/ShallowWrapper/context.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+expect(wrapper.context('foo')).to.equal(10);
+```
+
+
+#### Related Methods
+
+- [`.props() => Object`](props.md)
+- [`.prop(key) => Any`](prop.md)
+- [`.state(key) => Any`](state.md)
+- [`.state(key) => Any`](state.md)
```
Comment (lelandrichardson) [inline docs/api/ShallowWrapper/context.md]: duplicate?

File: docs/api/ReactWrapper/state.md [hunk 3]
```diff
@@ -25,3 +25,5 @@ expect(wrapper.state('foo')).to.equal(10);
 
 - [`.props() => Object`](props.md)
 - [`.prop(key) => Any`](prop.md)
+- [`.context() => Object`](context.md)
+- [`.context(key) => Any`](context.md)
```
Comment (lelandrichardson) [inline docs/api/ReactWrapper/state.md]: you might want to follow what `.state()` does and instead just have `.context([key]) => Any`

**Comment by blainekasten** [comment]
Fixed!

**Comment by lelandrichardson** [comment]
@blainekasten remember to add the 2 lines in /docs/README.md or else gitbook won't add those pages to the docs site!

**Comment by blainekasten** [comment]
Done.

**Pull request merged**